sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/batch/BulkExecutorTest.java

```diff
@@ -133,6 +151,276 @@ public void executeBulk_cancel() throws InterruptedException {
         }
     }
 
+    @Test(groups = { "emulator" }, timeOut = TIMEOUT)
+    public void executeBulk_preserveOrdering() throws InterruptedException {
+        int totalRequest = 100;
+        this.container = createContainer(database);
+
+        List<CosmosItemOperation> cosmosItemOperations = new ArrayList<>();
+        String duplicatePK = UUID.randomUUID().toString();
+        String id = UUID.randomUUID().toString();
+        for (int i = 0; i < totalRequest; i++) {
+            if (i == 0) {
+                BatchTestBase.EventDoc eventDoc = new BatchTestBase.EventDoc(id, 2, 4, "type1",
```

Comment:
it probably better to use the TestItem - the container is created with /mypk as the partitionKey, but the eventDoc does not has the mypk property, will the doc created successfully or created with null as partitionKey?

Reply:
The eventDoc does has a json property for mypk and the documents do get created successfully.